### PR TITLE
Doc page for sidenav component

### PR DESCRIPTION
**Pull request by Jialecl**

### CODE REVIEW

File: website/pages/components/switch/index.tsx [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
     </>
   );
 };
 
 Index.getLayout = function getLayout(page: ReactElement) {
-  return <SwitchPageLayout>{page}</SwitchPageLayout>;
+  return <SidenavPageLayout>{page}</SidenavPageLayout>;
 };
 
 export default Index;
```
Comment (marcialfps) [inline website/pages/components/switch/index.tsx:21]: Why this change???

File: website/screens/components/sidenav/usage/SidenavUsagePage.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            navigation links accessible to the users.
+          </DxcText>
+          <DxcText>
+            Keep in mind the type of the devices that you are developing for, and handle the
+            behavior in a way that doesn&#39;t block or reduce the available space of the main area
+            in the application
+          </DxcText>
+          <DxcText>
+            Try to follow and order for the sidenav content and make use of hierarchy to
+            differentiate between a title and a link
```
Comment (marcialfps) [inline website/screens/components/sidenav/usage/SidenavUsagePage.tsx:25]: Add full stops to sentences.

File: website/screens/components/sidenav/usage/SidenavUsagePage.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      <DxcStack gutter="large">
+        <HeadingLink level={2}>Usage</HeadingLink>
+        <DxcList>
+          <DxcText>
+            Use the sidenav element to improve the discoverability of the application, making the
+            navigation links accessible to the users.
+          </DxcText>
+          <DxcText>
+            Keep in mind the type of the devices that you are developing for, and handle the
+            behavior in a way that doesn&#39;t block or reduce the available space of the main area
```
Comment (marcialfps) [inline website/screens/components/sidenav/usage/SidenavUsagePage.tsx:20]: Remove `&#39;` and use apostrophe.

File: website/screens/components/sidenav/specs/SidenavSpecsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,557 @@
...
+          <DxcText>Title</DxcText>
+          <DxcText>Subtitle</DxcText>
+          <DxcText>Link</DxcText>
+          <DxcText>Container</DxcText>
+          <DxcText>Arrow</DxcText>
+        </DxcList>
+      </DxcStack>
+      <DxcStack gutter="large">
+        <HeadingLink level={3}>Design tokens</HeadingLink>
+        <HeadingLink level={4}>Design tokens</HeadingLink>
```
Comment (marcialfps) [inline website/screens/components/sidenav/specs/SidenavSpecsPage.tsx:33]: Wrong title. Change to `Color`.

File: website/screens/components/sidenav/specs/SidenavSpecsPage.tsx [hunk 5]
```diff
@@ -0,0 +1,557 @@
...
+          reduced, the default behavior in the first load of the page will be hidden in the sidenav
+          component. Taking this approach, as a first view of the page the user has all the content
+          in the main area visible, and then he will need to interact with the component to make it
+          visible and navigate to other resources.
+        </DxcText>
+        <DxcText as="p">
+          Also, as an important point to mention, the only variant that works with responsive is the
+          overlay, due to lack of space in the screen if the sidenav pushed the content of the main
+          container.
+        </DxcText>
```
Comment (marcialfps) [inline website/screens/components/sidenav/specs/SidenavSpecsPage.tsx:509]: This should be moved to Usage page like in the Header component.

File: website/screens/components/sidenav/specs/SidenavSpecsPage.tsx [hunk 6]
```diff
@@ -0,0 +1,557 @@
...
+            <tr>
+              <td>Arrow display</td>
+              <td>true (it cannot be false)</td>
+            </tr>
+            <tr>
+              <td>Arrow distance</td>
+              <td>50%</td>
+            </tr>
+          </tbody>
+        </DxcTable>
```
Comment (marcialfps) [inline website/screens/components/sidenav/specs/SidenavSpecsPage.tsx:492]: This table is repeated twice.

**Review by marcialfps — CHANGES_REQUESTED** [review]

File: website/screens/components/sidenav/usage/SidenavUsagePage.tsx (hunk 3 above)
Comment (Jialecl) [reply website/screens/components/sidenav/usage/SidenavUsagePage.tsx:20]: This is the same as an apostrophe and avoids triggering an error in visual studio.

File: website/screens/components/sidenav/specs/SidenavSpecsPage.tsx (hunk 5 above)
Comment (marcialfps) [reply website/screens/components/sidenav/specs/SidenavSpecsPage.tsx]: I think the table should be on the Specifications page and the text on the Usage page. What do you think @aweell ?

File: website/screens/components/sidenav/usage/SidenavUsagePage.tsx (hunk 3 above)
Comment (marcialfps) [reply website/screens/components/sidenav/usage/SidenavUsagePage.tsx:22]: It doesn't trigger any error in VS.
image

File: website/screens/components/sidenav/specs/SidenavSpecsPage.tsx (hunk 5 above)
Comment (aweell) [reply website/screens/components/sidenav/specs/SidenavSpecsPage.tsx]: Agree, the last paragraph and the table should live at the end of the specifications tab.

File: website/screens/components/sidenav/usage/SidenavUsagePage.tsx (hunk 3 above)
Comment (Jialecl) [reply website/screens/components/sidenav/usage/SidenavUsagePage.tsx:22]: I don't think this should be changed since it works the same way and also avoids errors in some VS.
Also it seems that there are multiple cases in which the code is used over the apostrophe.

**Review by marcialfps — APPROVED** [review]
